src/client.js

```diff
@@ -166,11 +262,18 @@ export class Saturn {
     }
 
     let fallbackCount = 0
-    for (const origin of this.nodes) {
+    const nodes = this.nodes
+    for (let i = 0; i < nodes.length; i++) {
       if (fallbackCount > this.opts.fallbackLimit) {
         return
       }
-      opts.url = origin.url
+      if (opts.raceNodes) {
+        const origins = nodes.slice(i, i + Saturn.defaultRaceCount).map((node) => node.url)
```

Comment:
So if we have nodes a,b,c,d,e, then the race groups are:

* a,b,c
* 1st fallback: b,c,d
* 2nd fallback: c,d,e

Is that right?

Reply:
That is correct. Will also note that this is not necessarily the best way to do it and has some obvious room for improvement. For example:
- If node c fails, it we might hit it twice again before another good fallback is contacted. 

I think as a next step, the best way to do this is to remove nodes that fail and such that:
- initial group is a, b, c. Assume node c fails
- 1st fallback is a, b, d and so on

I think this can be better achieved with a hashring implementation but we can re-evaluate.